x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/type/DataType.java

```diff
@@ -32,6 +32,90 @@
+ * Use capabilities to gate tests as you go, and use as many PRs as you think
+ * appropriate. New data types are complex, and smaller PRs will make reviews
+ * easier.
+ * <ul>
+ *     <li>Create a new feature flag for the type in {@link EsqlCorePlugin}.  We
+ *     recommend developing the data type over a series of smaller PRs behind
+ *     a feature flag. even for relatively simple data types.</li>
+ *     <li>Add a capability to EsqlCapabilities related to the new type, and
+ *     gated by the feature flag you just created.  Again, using the feature
+ *     flag is preferred over snapshot-only.  As development progresses, you may
+ *     need to add more capabilities related to the new type, e.g. for
+ *     supporting specific functions.  This is fine, and expected.</li>
+ *     <li>Create a new CSV test file for the new type.  Depending on the type,
+ *     you may also need to create a new sample data file if none of the
+ *     existing sample data files have fields of that type. See
+ *     CsvTestDataLoader for creating a new data set.</li>
+ *     <li>In the new CSV test file, start adding basic functionality tests.
+ *     These should include reading and returning values, both from indexed data
+ *     and from the ROW command.  It should also include functions that support
+ *     "every" type, such as MvFirst.</li>
+ *     <li>Add the new type to the CsvTestUtils#Type enum, if it isn't already
+ *     there.  You may also need to modify CsvAssert to support reading values
+ *     of the new type.</li>
+ *     <li>At this point, the CSV tests should fail with a sensible ES|QL error
+ *     message. Make sure they're failing in ES|QL, not in the test
+ *     framework.</li>
+ *     <li>Add the new data type to this enum.  This will cause a bunch of
+ *     compile errors for switch statements throughout the code.  Resolve those
+ *     as appropriate. That is the main way in which the new type will be tied
+ *     into the framework.</li>
+ *     <li>Add the new type to the {@link DataType#UNDER_CONSTRUCTION}
+ *     collection. This is used by the test framework to disable some checks
+ *     around how functions report their supported types, which would otherwise
+ *     generate a lot of noise while the type is still in development.</li>
+ *     <li>Add typed data generators to TestCaseSupplier, and make sure all
+ *     functions that support the new type have tests for it.</li>
+ *     <li>Work to support things all types should do.  Equality and the
+ *     "typeless" MV functions (MvFirst, MvLast, and MvCount) should work for
+ *     most types.  If the type has a natural ordering, make sure to test
+ *     sorting and the other binary comparisons. Make sure these functions all
+ *     have CSV tests that run against indexed data.</li>
+ *     <li>Add conversion functions as appropriate.  Almost all types should
+ *     support TO_STRING, and should have a TO_TYPE function that accepts a
```

Comment:
Just to stay consistent with how we named the other functions so far.

```suggestion
 *     support ToString, and should have a "ToType" function that accepts a
```